lib/logstash/filters/http.rb

```diff
@@ -93,7 +93,9 @@ def filter(event)
     else
       @logger.debug? && @logger.debug('success received',
                                       :code => code, :body => response_body)
-      process_response(response_body, response_headers, event)
+#       Since no response body will be present for HEAD request
+      if @verb != 'HEAD'
+        process_response(response_body, response_headers, event)
```

Comment:
While a HEAD request won't have a body, it can still have headers, which should be stored in the event. So instead of performing this check here, we can move it to `process_response()`

```ruby
  def process_response(body, headers, event)
    content_type, _ = headers.fetch("content-type", "").split(";")
    event.set(@target_headers, headers)
    return if @verb == 'HEAD' # there should be no body, we can return early
    if content_type == "application/json"
      begin
        parsed = LogStash::Json.load(body)
        event.set(@target_body, parsed)
```

Reply:
Hi @jsvd, yeah I missed that thing(headers we have to anyway set). Anyways updated the PR. Please have a look